azurerm/resource_arm_automation_dsc_configuration.go

```diff
@@ -0,0 +1,183 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/services/automation/mgmt/2015-10-31/automation"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAutomationDscConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAutomationDscConfigurationCreateUpdate,
+		Read:   resourceArmAutomationDscConfigurationRead,
+		Update: resourceArmAutomationDscConfigurationCreateUpdate,
+		Delete: resourceArmAutomationDscConfigurationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile(`^[a-zA-Z0-9_]{1,64}$`),
+					`The name length must be from 1 to 64 characters. The name can only contain letters, numbers and underscores.`,
+				),
+			},
+
+			"automation_account_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"location": locationSchema(),
+
+			"log_verbose": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"content_embedded": {
+				Type:         schema.TypeString,
+				Required:     true,
```

Comment:
*minor* Convention: Typically we sort the fields in "Required -> Optional -> Computed" order.

Reply:
Done